starlite/handlers/http.py

```diff
@@ -309,6 +312,8 @@ def __init__(
             response_cookies: A list of [Cookie](starlite.datastructures.Cookie] instances.
             response_headers: A string keyed dictionary mapping [ResponseHeader][starlite.datastructures.ResponseHeader]
                 instances.
+            responses: A dictionary of additional status codes and a description of their expected content (which is
+                assumed to be 'application/json'). This information will be included in the OpenAPI schema
```

Comment:
Do we need to assume json? You could have `AdditionalResponse.media_type` that defaults to `MediaType.JSON` but would allow customization if required.

Reply:
+1, media type should be specified per response type:

https://spec.openapis.org/oas/latest.html#responses-object

The thing is this, we need to be able to account for a use case that looks like this:


```python
@get("/some-path/{num: int}")
def handler(num: int) -> Response:
    if num == 1:
       return Response(media_type=MediaType.JSON, status_code=HTTP_200_OK, content={"hello": "world"})
    elif num == 2:
        return Response(media_type=MediaType.TEXT, status_code=HTTP_202_ACCEPTED, content="hello world")
    return Response(media_type=MediaType.HTML, status_code=HTTP_201_CREATED, content="<html><body><div>hello world</div></body></html>")
```

That is, the user can dynamically decide to return different media types, different status codes etc.

Reply:
Addressed in https://github.com/starlite-api/starlite/pull/478/commits/dc2b02e52214f4ef2be388ce4d5f114615bea6bb

Reply:
Also fixed `responses` documentation in https://github.com/starlite-api/starlite/pull/478/commits/8450baea0643b40be4c6332a0e894a32d01855d8